<a href="https://colab.research.google.com/github/Roshanshres/predict-diabetes-svm-/blob/main/diabeties_using_svm.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [4]:
df = pd.read_csv("/content/diabetes.csv")

In [5]:
df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [6]:
df.tail()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0

In [7]:
df.head(2)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0

In [9]:
df.shape

(768, 9)

In [10]:
#sampling
df.sample()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
376            0       98             82             15       84  25.2   

     DiabetesPedigreeFunction  Age  Outcome  
376                     0.299   22        0

In [11]:
df.sample(6)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
445            0      180             78             63       14  59.4   
232            1       79             80             25       37  25.4   
286            5      155             84             44      545  38.7   
706           10      115              0              0        0   0.0   
282            7      133             88             15      155  32.4   
303            5      115             98              0        0  52.9   

     DiabetesPedigreeFunction  Age  Outcome  
445                     2.420   25        1  
232                     0.583   22        0  
286                     0.619   34        0  
706                     0.261   30        1  
282                     0.262   37        0  
303                     0.209   28        1

In [12]:
df.sample(n=5, random_state =40)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
370            3      173             82             48      465  38.4   
388            5      144             82             26      285  32.0   
611            3      174             58             22      194  32.9   
550            1      116             70             28        0  27.4   
232            1       79             80             25       37  25.4   

     DiabetesPedigreeFunction  Age  Outcome  
370                     2.137   25        1  
388                     0.452   58        1  
611                     0.593   36        1  
550                     0.204   21        0  
232                     0.583   22        0

checking datatypes if columns

In [14]:
df.dtypes

Pregnancies                   int64
Glucose                       int64
BloodPressure                 int64
SkinThickness                 int64
Insulin                       int64
BMI                         float64
DiabetesPedigreeFunction    float64
Age                           int64
Outcome                       int64
dtype: object

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [16]:
print(type(df))

<class 'pandas.core.frame.DataFrame'>




*   Dara contains 768 rows or entities
*   out of 768 rows each column has no ant not null or missing data


In [17]:
df.isna().sum() #no missing row in a data

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64